src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptions.cs

```diff
@@ -25,13 +35,42 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The selected persistence doesn't have support for subscription storage. Select another persistence or disable the message-driven subscriptions feature using endpointConfiguration.DisableFeature<MessageDrivenSubscriptions>()");
             }
 
-            context.Pipeline.Register<SubscriptionReceiverBehavior.Registration>();
-            var authorizer = context.Settings.GetSubscriptionAuthorizer();
-            if (authorizer == null)
+            var transportInfrastructure = context.Settings.Get<TransportInfrastructure>();
+            var canReceive = !context.Settings.GetOrDefault<bool>("Endpoint.SendOnly");
+            var conventions = context.Settings.Get<Conventions>();
+            var enforceBestPractices = context.Settings.Get<bool>(BestPracticesFeature.EnforceBestPracticesSettingsKey);
```

Comment:
Would it be possible / feasible to move this to best practices feature? For example the apply method could just take the publisher and the conventions and then the best practices feature goes over all registered publishers and sets the flag later or checks the registered once? Of course this would mean another feature dependency :( But then we would get rid of the best practices here and move all related code into the feature it belongs